### PR TITLE
Add shim for Spark 3.0.3

### PR DESCRIPTION
**Pull request by jlowe**
Spark 3.0.3 is starting to publish snapshots, so adding a shim that can support that version.

### CODE REVIEW

**Review by tgravescs — DISMISSED** [review]

**Comment by jlowe** [comment]
build

**Review by gerashegalov — DISMISSED** [review]
LGTM

**Review by gerashegalov — APPROVED** [review]

**Comment by jlowe** [comment]
build

File: jenkins/spark-nightly-build.sh [hunk 1]
```diff
@@ -25,6 +25,7 @@ mvn -U -B -Pinclude-databricks,snapshot-shims clean deploy $MVN_URM_MIRROR -Dmav
 # Run unit tests against other spark versions
 mvn -U -B -Pspark301tests,snapshot-shims test $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR
 mvn -U -B -Pspark302tests,snapshot-shims test $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR
+mvn -U -B -Pspark303tests,snapshot-shims test $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR
```
Comment (revans2) [inline jenkins/spark-nightly-build.sh:28]: Do we want the pre-commit build to start testing 303 too?

**Review by revans2 — APPROVED** [review]

File: jenkins/spark-nightly-build.sh (hunk 1 above)
Comment (jlowe) [reply jenkins/spark-nightly-build.sh:28]: Yes, I think that's a good thing.  I noticed we're not testing against spark302 in premerge, shall I add that while I'm at it?

Comment (revans2) [reply jenkins/spark-nightly-build.sh:28]: I think so. We really need to get to the point where we are dropping support for bugfix releases, but until then we need to test what we support. And it should only add 7 to 10 mins per version to the runtime of the premerge build.  Not ideal but acceptable for now.

**Comment by jlowe** [comment]
build

**Comment by jlowe** [comment]
build

**Review by tgravescs — APPROVED** [review]

**Pull request merged**